board/xilinx/zynqmp/zynqmp.c

```diff
@@ -965,3 +965,32 @@ int spl_mmc_get_uboot_payload_filename(char *filename, size_t len,
 	return 0;
 }
 #endif
+
+#if defined(CONFIG_SPL_SPI_SUPPORT)
+unsigned int spl_spi_get_uboot_offs(struct spi_flash *flash)
+{
+	int ret;
+	u32 multiboot;
+	u32 spi_offset = 0;
+
+	multiboot = multi_boot_get();
+
+	switch(multiboot) {
+	case 0:
+		spi_offset = CONFIG_SYS_SPI_U_BOOT_OFFS;
+		break;
+	case 10:
```

Comment:
The question I asked yesterday
@ricardosalveti 
@ldts @mike-scott 

are the these offsets OK?

Primary boot.bin - `0x0`
Secondary boot.bin - `0x50000`  (corresponds to multiboot index `10`, as `10*32KB=0x50000`)
Primary U-Boot FIT - `0x100000`
Secondary U-Boot FIT - `0xaa0000`

Also yesterday @ricardosalveti mentioned the third set of images? Could you please elaborate?

Reply:
I would let enough space to allow the usage of a third/golden image, which can later be used for recovery. Just because this board is a bit special that jtag is completely disable once RSA_EN is fused, so the only way to recover from a broken primary/secondary is to make sure there is always a golden that can at least enable jtag again.

Reply:
Not an issue with avnet as the boot switches are exposed, but boards booting from qspi and without boot switches can easily get in a bad state.